### ISSUE TITLE
ComputeUsagesTask taking 30+ minutes to execute when it used to take 30 seconds for the same project

### ISSUE THREAD

**Issue by kyhule** [autonomousapps/dependency-analysis-gradle-plugin#1136]
**Build scan link**
I have buildscans on company's internal instance if need info.
![CleanShot 2024-02-20 at 12 55 05@2x](https://github.com/autonomousapps/dependency-analysis-gradle-plugin/assets/2845460/9f7f0742-787a-4c9e-a3a0-6af313424895)
![CleanShot 2024-02-20 at 12 55 27@2x](https://github.com/autonomousapps/dependency-analysis-gradle-plugin/assets/2845460/5a180843-9377-4e5c-9cd4-5afe45512b8d)

**Plugin version**
Seeing in 1.29.0 and 1.30.0

**Gradle version**
8.6

**JDK version**
17

**(Optional) Kotlin and Kotlin Gradle Plugin (KGP) version**
1.9.20

**(Optional) Android Gradle Plugin (AGP) version**
8.1.4

**Describe the bug**
When running `projectHealth`, `ComputeUsagesTask` increased from ~30s to ~40m after upgrading to version 1.29.0 on our main `app` module.

**To Reproduce**
1. ...

**Expected behavior**

**Additional context**

**Comment by autonomousapps** [comment]
Are you using the configuration cache?

**Comment by kyhule** [comment]
No, configuration cache was not enabled for either run. No, configuration cache was not enabled for either run.

**Comment by autonomousapps** [comment]
Can you try 1.30.0?

If you get the same results, I'd like to see CPU and memory profiles to investigate. The biggest change between 28 and 29 is using the new AGP APIs. 30 makes further improvements in this vein. If you're not already doing it, I would suggest disabling auto-apply behavior and instead manually applying DAGP to each module.

**Comment by kyhule** [comment]
I saw the same results using 1.30.0 which is where I first noticed it. I dropped back to 1.29.0 first and then 1.28.0 before the regression went away. I am relying on the auto apply; I'll try running without it.  I am upgrading to AGP 8.2.2 right now so I'll see if that makes any difference. If none of that proves to help, I'll get cpu and memory usage on it.

**Comment by autonomousapps** [comment]
I wonder if memory usage went up and you're GC thrashing. Could you take a look in the performance tab of your build scans and see what's happening with your memory?

**Comment by kyhule** [comment]
> I wonder if memory usage went up and you're GC thrashing. Could you take a look in the performance tab of your build scans and see what's happening with your memory?
Nope, doesn't seem to be a GC issue.
<img width="579" alt="CleanShot 2024-02-24 at 15 16 32@2x" src="https://github.com/autonomousapps/dependency-analysis-gradle-plugin/assets/2845460/fd29f1ed-1de1-4c7b-ace6-062a6fa8c671">

**Comment by kyhule** [comment]
I wonder if it is due to the fact that we have `nonTransitiveRClass` disabled (set to `false`). Did a heap dump and seeing a ton of instances of AndroidRes objects.
<img width="954" alt="CleanShot 2024-02-24 at 15 26 19@2x" src="https://github.com/autonomousapps/dependency-analysis-gradle-plugin/assets/2845460/9eb8f10d-75a2-4226-a771-3a80f629e80b">

**Comment by autonomousapps** [comment]
> I wonder if it is due to the fact that we have `nonTransitiveRClass` disabled (set to `false`). Did a heap dump and seeing a ton of instances of AndroidRes objects. <img alt="CleanShot 2024-02-24 at 15 26 19@2x" width="954" src="https://private-user-images.githubusercontent.com/2845460/307548078-9eb8f10d-75a2-4226-a771-3a80f629e80b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDg4MTMwNDUsIm5iZiI6MTcwODgxMjc0NSwicGF0aCI6Ii8yODQ1NDYwLzMwNzU0ODA3OC05ZWI4ZjEwZC03NWEyLTQyMjYtYTc3MS0zYTgwZjYyOWU4MGIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDIyNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAyMjRUMjIxMjI1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YTI5MGU1NDBkMzU1ZDYzMTU5NWU5MDdmNGU3YmIwY2E5ZTI0OWRmZTdlMjJlMTcyNWU5ZGE2MDllYjYwY2I3NCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.SAD9QYcokpRXzr801Lm97FxDBimU5gAcpJY8zetom28">

maybe. the plugin can't even give accurate advice relating to Android resources with that disabled.
